ipa-core/src/net/server/handlers/query/create.rs

```diff
@@ -48,50 +47,39 @@ mod tests {
         helpers::{
             make_owned_handler,
             query::{IpaQueryConfig, PrepareQuery, QueryConfig, QueryType},
-            routing::{Addr, RouteId},
-            HelperIdentity, HelperResponse, Role, RoleAssignment,
+            routing::RouteId,
+            HelperResponse, Role, RoleAssignment,
         },
         net::{
             http_serde,
-            server::handlers::query::test_helpers::{assert_req_fails_with, IntoFailingReq},
-            test::TestServer,
+            server::handlers::query::test_helpers::{assert_fails_with, assert_success_with},
         },
         protocol::QueryId,
     };
 
     async fn create_test(expected_query_config: QueryConfig) {
-        let test_server = TestServer::builder()
-            .with_request_handler(make_owned_handler(
-                move |addr: Addr<HelperIdentity>, _| async move {
-                    let RouteId::ReceiveQuery = addr.route else {
-                        panic!("unexpected call");
-                    };
-
-                    let query_config = addr.into().unwrap();
-                    assert_eq!(query_config, expected_query_config);
-                    Ok(HelperResponse::from(PrepareQuery {
-                        query_id: QueryId,
-                        config: query_config,
-                        roles: RoleAssignment::try_from([Role::H1, Role::H2, Role::H3]).unwrap(),
-                    }))
-                },
-            ))
-            .build()
-            .await;
-        let req = http_serde::query::create::Request::new(expected_query_config);
-        let req = req
+        let req = http_serde::query::create::Request::new(expected_query_config)
             .try_into_http_request(Scheme::HTTP, Authority::from_static("localhost"))
             .unwrap();
-        let resp = test_server.server.handle_req(req).await;
-
-        let status = resp.status();
-        let body_bytes = hyper::body::to_bytes(resp.into_body()).await.unwrap();
-        let response_str = String::from_utf8(body_bytes.to_vec()).unwrap();
-
-        assert_eq!(StatusCode::OK, status, "Request failed: {}", &response_str);
+        let handler = make_owned_handler(
+            // Do we need these moves? In particular the first one?
```

Comment:
yea, we do because this thing needs to be thread-safe and can't have any references to outside.